src/NHibernate/Dialect/Schema/DB2MetaData.cs

```diff
@@ -31,13 +32,20 @@ public override ISet<string> GetReservedWords()
 				result.Add(row["ReservedWord"].ToString());
 			}
 
-			if (IncludeDataTypesInReservedWords)
+			if (!IncludeDataTypesInReservedWords)
+				return result;
+
+			var dtTypes = Connection.GetSchema(DbMetaDataCollectionNames.DataTypes);
+
+			var typeNameColumn = dtTypes.Columns.Cast<DataColumn>()
```

Comment:
How many columns are returned?

I believe this check can be simplified with `dtTypes.Columns.Contains("SQL_TYPE_NAME")`. Or  `dtTypes.Columns.IndexOf` and refactor  to use index instead.

Reply:
23 columns in my debug :

![image](https://github.com/nhibernate/nhibernate-core/assets/21284787/bdf16f1c-ecce-49c7-b11a-c9a8200026ff)

Just pushed a new version that uses IndexOf. It will also fallback to "TypeName" if "SQL_TYPE_NAME" is not found.